test/gtest/uct/test_md.cc

```diff
@@ -670,48 +701,60 @@ UCS_TEST_SKIP_COND_P(test_md, reg_advise,
-                     !check_caps(UCT_MD_FLAG_ALLOC |
-                                 UCT_MD_FLAG_ADVISE)) {
-    uct_md_h md_ref           = md();
-    uct_alloc_method_t method = UCT_ALLOC_METHOD_MD;
-    void *address             = NULL;
-    size_t size, orig_size;
+static void test_alloc_advise(uct_md_h md_ref, ucs_memory_type_t mem_type)
+{
+    constexpr size_t orig_size          = 128 * UCS_MBYTE;
+    constexpr size_t advise_size        = 32 * UCS_KBYTE;
+    constexpr uct_alloc_method_t method = UCT_ALLOC_METHOD_MD;
+    void *address                       = NULL;
+    size_t size;
     ucs_status_t status;
     uct_allocated_memory_t mem;
     uct_mem_alloc_params_t params;
 
-    params.field_mask      = UCT_MEM_ALLOC_PARAM_FIELD_FLAGS    |
-                             UCT_MEM_ALLOC_PARAM_FIELD_ADDRESS  |
-                             UCT_MEM_ALLOC_PARAM_FIELD_MEM_TYPE |
-                             UCT_MEM_ALLOC_PARAM_FIELD_MDS      |
-                             UCT_MEM_ALLOC_PARAM_FIELD_NAME;
-    params.flags           = UCT_MD_MEM_FLAG_NONBLOCK | UCT_MD_MEM_ACCESS_ALL;
-    params.name            = "test";
-    params.mem_type        = UCS_MEMORY_TYPE_HOST;
-    params.address         = address;
-    params.mds.mds         = &md_ref;
-    params.mds.count       = 1;
-
-    size          = 128 * UCS_MBYTE;
-    orig_size     = size;
-
-    status  = uct_mem_alloc(size, &method, 1, &params, &mem);
+    params.field_mask = UCT_MEM_ALLOC_PARAM_FIELD_FLAGS |
+                        UCT_MEM_ALLOC_PARAM_FIELD_ADDRESS |
+                        UCT_MEM_ALLOC_PARAM_FIELD_MEM_TYPE |
+                        UCT_MEM_ALLOC_PARAM_FIELD_MDS |
+                        UCT_MEM_ALLOC_PARAM_FIELD_NAME;
+    params.flags      = UCT_MD_MEM_FLAG_NONBLOCK | UCT_MD_MEM_ACCESS_ALL;
+    params.name       = "test";
+    params.mem_type   = mem_type;
+    params.address    = address;
+    params.mds.mds    = &md_ref;
+    params.mds.count  = 1;
+
+    status  = uct_mem_alloc(orig_size, &method, 1, &params, &mem);
     address = mem.address;
     size    = mem.length;
     ASSERT_UCS_OK(status);
     EXPECT_GE(size, orig_size);
     EXPECT_TRUE(address != NULL);
     EXPECT_TRUE(mem.memh != UCT_MEM_HANDLE_NULL);
 
-    status = uct_md_mem_advise(md(), mem.memh, (char *)address + 7,
-                               32 * UCS_KBYTE, UCT_MADV_WILLNEED);
+    status = uct_md_mem_advise(md_ref, mem.memh, (char*)address + 7,
+                               advise_size, UCT_MADV_WILLNEED);
     EXPECT_UCS_OK(status);
 
     memset(address, 0xBB, size);
     uct_mem_free(&mem);
 }
 
+UCS_TEST_SKIP_COND_P(test_md, alloc_advise,
+                     !check_caps(UCT_MD_FLAG_ALLOC | UCT_MD_FLAG_ADVISE))
+{
+    uct_md_h md_ref = md();
+    ucs_memory_type_t mem_type;
+    uint32_t mem_type_index;
+
+    ucs_for_each_bit(mem_type_index, md_attr().alloc_mem_types) {
+        if (md_attr().access_mem_types & UCS_BIT(mem_type_index)) {
```

Comment:
why needed?

Reply:
to avoid testing advise for memory types which won't support it such as memic